src/common/src/row/project.rs

```diff
@@ -82,6 +82,21 @@ impl<R: Row> Hash for Project<'_, R> {
     }
 }
 
+impl<R: Row> std::fmt::Display for Project<'_, R> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
This appears to be applicable to all `impl Row`s.

May make it a provided method of `Row` trait and forward the implementation here.

Reply:
Reference: https://github.com/risingwavelabs/risingwave/pull/12652/files#diff-fed9965020a045a06e045983438c3903b3b9debb7a18de4d641793d5e917cb15R114-R121